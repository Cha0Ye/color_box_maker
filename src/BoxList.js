import React, { Component } from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import uuid from 'uuid/v4';

class BoxList extends Component {
    constructor(props){
        super(props)
        this.state = {
            boxes: []
        };
    }


    render() {
        return (
            <div>
                {this.state.boxes.Map( b =>  <Box boxStyle={ b } triggerDelete={FIXME FUNCTION}/>)}
                <NewBoxForm triggerCreate={ FIXME FUNCTION}/>
                
            </div>
        );
    }
}

export default BoxList;