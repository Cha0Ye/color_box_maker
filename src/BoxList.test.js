import React from 'react';
import BoxList from './BoxList';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

let wrapper;

beforeEach(() => {
    wrapper = mount(<BoxList />);
})

// smoke test
it('renders without crashing', () => {
    const wrapper = shallow(<BoxList />);
});

// snapshot test
it('matches snapshot', () => {

    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

it("adds boxes", function () {
    const items = [{ id: 1, width: '20px', height: '20px', backgroundColor: 'purple' }];
    const wrapper = mount(<BoxList />);

    wrapper.setState({ boxes: items });
    expect(wrapper.state().boxes).toEqual([
        { id: 1, width: '20px', height: '20px', backgroundColor: 'purple' }
    ]);

    // here we use the "instance" method
    // to get access to all instance methods defined on the component
    wrapper
        .instance()
        .create({ width: '40px', height: '40px', backgroundColor: 'blue' });

    expect(wrapper.state().boxes).toEqual([
        { id: 1, width: '20px', height: '20px', backgroundColor: 'purple' },
        { id: expect.any(String), width: '40px', height: '40px', backgroundColor: 'blue' }
    ]);
});

it("finds a form and successfully creates ", function() {
    const wrapper = mount(<BoxList />);

    const widthInput = wrapper.find("#width")
    widthInput.instance().value = "20px"
    widthInput.simulate("change")

    // expect(wrapper.state().width).toEqual("20px");

    const heightInput = wrapper.find("#height");
    heightInput.instance().value = '20px';
    heightInput.simulate("change");

    // expect(wrapper.state().height).toEqual('20px');

    const backgroundColorInput = wrapper.find("#backgroundColor");
    backgroundColorInput.instance().value = 'blue';
    backgroundColorInput.simulate("change");

    // expect(wrapper.state().backgroundColor).toEqual('blue');


    const form = wrapper.find("form");
    form.simulate("submit");

    expect(wrapper.state().boxes.length).toEqual(1);

    // since we are appending to the end, find the last item,
    // and make sure it matches what we have just created
    expect(
      wrapper.find("div").last().text()
    ).toEqual('X');
  });
