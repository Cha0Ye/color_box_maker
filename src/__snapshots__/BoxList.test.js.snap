// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<BoxList>
  <div>
    <NewBoxForm
      triggerCreate={[Function]}
    >
      <div>
        <form
          onSubmit={[Function]}
        >
          <label
            htmlFor="width"
          >
            width: 
          </label>
          <input
            id="width"
            name="width"
            onChange={[Function]}
            value=""
          />
          <label
            htmlFor="height"
          >
            Height: 
          </label>
          <input
            id="height"
            name="height"
            onChange={[Function]}
            value=""
          />
          <label
            htmlFor="backgroundColor"
          >
            Background Color: 
          </label>
          <input
            id="backgroundColor"
            name="backgroundColor"
            onChange={[Function]}
            value=""
          />
          <button>
            CreateBox
          </button>
        </form>
      </div>
    </NewBoxForm>
    <div />
  </div>
</BoxList>
`;
